### ISSUE TITLE
Refactor patch-sources.sh used by the Operator

### ISSUE THREAD

**Issue by stianst** [keycloak/keycloak#15109]
### Description

The Operator currently uses a bash script (with sed behind the scenes) to modify Representation classes from the core module, by adding a number of additional Fabric8 annotations.

It is problematic that this is a bash script using the maven-exec-plugin from a portability perspective (Linux, Mac, Windows, etc.). Further, it is also problematic and can have unwanted side-effects that it effectively adds modified versions of the representation classes onto the classpath.

This approach needs to be completely rethought/reworked as it's not something we can keep in the long run.

**Comment by stianst** [comment]
We had a number of developers already waste a significant amount of time trying to get Keycloak imported into IntelliJ due to this approach. With errors like:

```
java: java.lang.IllegalArgumentException: Found a cyclic reference involving the field subGroups of type org.keycloak.representations.idm.GroupRepresentation
```

When I encountered this issue I had no clue what was going on, or how to resolve this problem, which in fact it's always going to be broken unless we re-work it as there will be multiple versions of representation classes on the class-path when the Operator module is enabled.

**Comment by vmuzikar** [comment]
Duplicate of #10736

**Comment by stianst** [comment]
#10736 might be one solution to this issue. Is that still blocked by Fabric8 being updated? Are there any other options that we can look at? This really is a big time-waster for folks and something we should look at improving asap.

**Comment by vmuzikar** [comment]
IMHO #10736 is the correct way we should go as it aligns the native approach to generate CRDs. Anything else would be just another workaround.

And +1 that this should be prioritized.

In the meantime I created #15111 to unblock folks who were hitting this issue.

**Comment by shawkins** [comment]
Closed by #10736
